src/sentry/static/sentry/app/components/actions/resolve.tsx

```diff
@@ -9,8 +9,42 @@ import ActionLink from 'app/components/actions/actionLink';
 import Tooltip from 'app/components/tooltip';
 import GuideAnchor from 'app/components/assistant/guideAnchor';
 import {formatVersion} from 'app/utils/formatters';
+import {Release} from 'app/types';
 
-export default class ResolveActions extends React.Component {
+type StatusDetails = {
+  inRelease?: string;
+  inNextRelease?: boolean;
+};
+
+enum ResolveStatus {
+  RESOLVED = 'resolved',
+  UNRESOLVED = 'unresolved',
+}
+
+const defaultProps = {
+  isResolved: false,
+  isAutoResolved: false,
+  confirmLabel: t('Resolve'),
+};
+
+type Props = {
+  hasRelease: boolean;
+  onUpdate: (params: {status: ResolveStatus; statusDetails?: StatusDetails}) => void;
```

Comment:
Should this be a type that is exported? Will parent components need this type too?

Reply:
Yes, will be handy to have it exported when converting `actions.jsx`, thanks